### PR TITLE
Add SQL tools dsq, octosql, sqls, sql-language-server

### PR DESCRIPTION
**Pull request by danielnachun**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by danielnachun** [comment]
@conda-forge/staged-recipes This is now ready for review.  I had to skip the windows build for two package:

- `sql-language-server` builds successfully, but hits the well known issue of file paths being too long: https://github.com/conda/conda/issues/7203.  The solutions I tried there did not work so far.
- `sqls` is a CGO package and does not work with MSVC.  I had no success getting to work with MinGW, though this may be easier to iterate on once we have a feedstock.

File: recipes/sqls/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+c_stdlib_version:  # [osx]
+  - "10.15"  # [osx]
+MACOSX_SDK_VERSION:  # [osx]
+  - "10.15"  # [osx]
```
Comment (xhochy) [inline recipes/sqls/conda_build_config.yaml:4]: This should not apply to `osx-arm64`
```suggestion
c_stdlib_version:  # [osx and x86_64]
  - "10.15"  # [osx and x86_64]
MACOSX_SDK_VERSION:  # [osx and x86_64]
  - "10.15"  # [osx and x86_64]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by danielnachun** [comment]
The requested changes have been made - see the last commit.

File: recipes/sqls/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+c_stdlib_version:  # [osx and x86_64]
+  - "10.15"  # [osx]
+MACOSX_SDK_VERSION:  # [osx and x86_64]
+  - "10.15"  # [osx]
```
Comment (xhochy) [inline recipes/sqls/conda_build_config.yaml:4]:
```suggestion
  - "10.15"  # [osx and x86_64]
MACOSX_SDK_VERSION:  # [osx and x86_64]
  - "10.15"  # [osx and x86_64]
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Still a bit missing

**Pull request merged**